### ISSUE TITLE
Invalid `git annex findref` call on Windows

### ISSUE THREAD

**Issue by mih** [datalad/datalad#7248]
```
CommandError: '"git" "-c" "diff.ignoreSubmodules=none" "annex" "findref" "--anything" "HEAD" "--json" "--json-error-messages" "-c" "annex.dotfiles=true"' failed with exitcode 1 under C:\DLTMP\datalad_temp_tree_em3s022d\ds [info keys: stdout_json] [err: 'Invalid option `--anything'
```

Seems to be due to a wrong version comparison result in datalad proper

**Comment by mih** [comment]
This change came with a57f461bbb, but the breakage was not detactable, because an old version is used for windows testing here. It would probably be sensible to adopt the setup from extension testing.

**Comment by bpoldrack** [comment]
The issue was introduced by PR #7230, which utilizes `git annex find --anything`. The switch was introduced in annex but not yet released, hence the version kludge specifies `10.20221212+git18` as requirement to use it.

However, `10.20221212` windows build of annex reports in fact `10.20221212-gab11fd70e` (Note, that `ab11fd70e` *is* `10.20221212`. As a result, datalad is determining that `10.20221212-gab11fd70e` >= `10.20221212+git18` and therefore ends up using `--anything` with what really is annex `10.20221212` on windows.

**Comment by yarikoptic-gitmate** [comment]
Issue fixed in [`0.18.1`](https://github.com/datalad/datalad/releases/tag/0.18.1)
